File: C_Programming/Day_7/program60.c
// Description : Write a program that accepts array size , target  , elements from the user ,  
//               & search  whether that element is present in that array or not .

#include<stdio.h>
#include<stdlib.h>
# include<stdbool.h>

bool FindElement(int *Arr , int iSize , int iTarget)
{
    int iCnt = 0;
    bool bFlag = false;
    

    for(iCnt = 0 ; iCnt < iSize ; iCnt++)
    {
        if(Arr[iCnt] == iTarget)
        {
            bFlag = true;
        }
    }

    return bFlag;
    
}

void Display(int * Arr , int iSize)
{
    int iCnt = 0;

    printf("Elements from the array are : \n");
    for(iCnt = 0 ; iCnt < iSize ; iCnt++)
    {
        printf("%d\t",Arr[iCnt]);
    }
    printf("\n");
}

void Accept(int * Arr , int iSize)
{
    int iCnt = 0;

    printf("Enter the elements : \n");
    for(iCnt = 0 ; iCnt < iSize ; iCnt++)
    {
        scanf("%d",&Arr[iCnt]);
    }
}

int main()
{
    int * Arr = NULL;
    int iLength = 0;
    bool bRet = false;
    int iTarget = 0;

    printf("Enter the size of the array : ");
    scanf("%d",&iLength);

    printf("Enter the number that you want to serach in the array : ");
    scanf("%d",&iTarget);

    Arr = (int *)malloc(sizeof(int) * iLength);

    Accept(Arr,iLength);
    Display(Arr,iLength);

    bRet = FindElement(Arr , iLength , iTarget);
    if(bRet == true)
    {
        printf("%d is present...\n",iTarget);
    }
    else
    {
        printf("%d is not present...\n",iTarget);
    }

    return 0;
}

File: Python_Programming/Day_7/program60.py
# Description:  Write a program that accepts array size , target  , elements from the user ,  
#               & search  whether that element is present in that array or not .

def Accept(Arr , Size):
    Value = 0

    print("Enter the elements : ")
    for i in range(Size):
      Value = int(input())
      Arr.append(Value) 

def Display(Arr , Size):

    print("Elements are : ")
    for i in range(Size):
        print(Arr[i] , "\t" , end = "")
    
    print("\n")

def FindElement(Arr , Size , Target):
    Flag = False

    for i in range(Size):
        if(Arr[i] == Target):
            Flag  = True

    return Flag

def main():
    Arr = list()
    Length = 0
    Ret = 0
    Target = 0

    print("Enter the size of the list : ",end = "")
    Length = int(input())

    print("Enter the number that you want to search : " , end = "")
    Target = int(input())

    Accept(Arr , Length)
    Display(Arr,Length)

    Ret = FindElement(Arr , Length , Target)
    if(Ret == True):
        print(f"{Target} is present...")
    else : 
        print(f"{Target} is not present...")





if __name__ == "__main__":
    main()